### PR TITLE
Closes CPPL-278

### PR DESCRIPTION
**Pull request by iodream**
Adds Fonts changing section in Settings. Change fonts by setting a size in spinbox or select a custom font in selector. 
Fonts are saved and read from settings.json but not applied when user is logged out. 
Screenshot_20220504_211805

### CODE REVIEW

**Review by SantaBar6arra — APPROVED** [review]

File: src/Client/Controller/Controller.cpp [hunk 1]
```diff
@@ -80,6 +81,10 @@ void Controller::ReadSettings()
 		MainPage::ColorSettings::LIST_INACTIVE = json.value("COLOR_LIST_INACTIVE").toString();
 		MainPage::ColorSettings::NAVBUTTONS = json.value("COLOR_NAVBUTTONS").toString();
 		MainPage::ColorSettings::RECOMMENDATION = json.value("COLOR_RECOMMENDATION").toString();
+
+		QString FontName = json.value("UI_FONT_NAME").toString();
+		int FontSize= json.value("UI_FONT_SIZE").toInt();
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/Controller.cpp]: Missing space

File: src/Client/Controller/Controller.cpp [hunk 2]
```diff
@@ -189,6 +193,9 @@ void Controller::OnSaveConfig()
 	json["COLOR_NAVBUTTONS"] = MainPage::ColorSettings::NAVBUTTONS;
 	json["COLOR_RECOMMENDATION"] = MainPage::ColorSettings::RECOMMENDATION;
 
+	json["UI_FONT_NAME"] = MainPage::UISettings::UI_FONT.family();
+	json["UI_FONT_SIZE"] = MainPage::UISettings::UI_FONT.pointSize();
+
 	QFile file(settings_filename);
 	QByteArray bytes = QJsonDocument(json).toJson( QJsonDocument::Indented );
```
Comment (yaroslav-kernitskyi) [inline src/Client/Controller/Controller.cpp]: Not your code, but strange formatting

File: src/Client/View/MainPage/MainPage.cpp [hunk 3]
```diff
@@ -282,12 +282,19 @@ void MainPage::ShowBalance(const Balance& money)
...
+void MainPage::UpdatePage()
 {
 	//balance banners
 	m_ui->BalanceHolder->setStyleSheet(QString("background-color:" + ColorSettings::COLOR_TOP_BANNER));
 
+	//general text and font size
+	setStyleSheet(QString("color:%1; font-family:%2; font-size:%3px")
+						  .arg(ColorSettings::LABEL_TEXT)
+						  .arg(MainPage::UISettings::UI_FONT.family())
+						  .arg(MainPage::UISettings::UI_FONT.pointSize()));
```
Comment (yaroslav-kernitskyi) [inline src/Client/View/MainPage/MainPage.cpp]: Formatting

**Review by yaroslav-kernitskyi — APPROVED** [review]

File: src/Client/Controller/Controller.cpp (hunk 2 above)
Comment (iodream) [reply src/Client/Controller/Controller.cpp]: do you mean the extra newlines? i can remove them, i just thought the font values could be grouped together from the color values

Comment (yaroslav-kernitskyi) [reply src/Client/Controller/Controller.cpp]: I meant these spaces
`.toJson( QJsonDocument::Indented );`

**Pull request merged**